ush/python/pygfs/task/atm_analysis.py

```diff
@@ -82,11 +82,17 @@ def initialize(self: Analysis) -> None:
         jedi_fix_list = parse_j2yaml(jedi_fix_list_path, self.task_config)
         FileHandler(jedi_fix_list).sync()
 
-        # stage berror files
-        # copy static background error files, otherwise it will assume ID matrix
+        # stage static background error files, otherwise it will assume ID matrix
         logger.debug(f"Stage files for STATICB_TYPE {self.task_config.STATICB_TYPE}")
         FileHandler(self.get_berror_dict(self.task_config)).sync()
 
+        # stage ensemble files for use in hybrid background error
+        if self.task_config.DOHYBVAR:
+            logger.debug(f"Stage ensemble files for DOHYBVAR {self.task_config.DOHYBVAR}")
+            self.task_config.RUN = 'enkf' + self.task_config.RUN
+            self.task_config.dirname ='ens'
```

Comment:
These are dangerous.
After these lines, one would have altered the `self.task_config`   This is forbidden.
What you should do here instead, is make a deepcopy of `self.task_config`, extract the attributes that you need to send to `self.get_fv3ens_dict`, instead of sending in the entire task configuration.

Reply:
I guess having this in `get_fv3ens_dict` was safer.  I could move it back but I don't think you want this.  I don't know how to do what you outline.  Is there an example somewhere in g-w that I can follow.

Reply:
see `land_analysis.py`.
You could follow something along these lines (not what this method needs), but an example of creating a local dict and pulling relevant info from `self.task_config`.
```
        localconf = AttrDict()
        keys = ['DATA', 'current_cycle', 'COM_OBS', 'COM_ATMOS_RESTART_PREV',
                'OPREFIX', 'CASE', 'ntiles']
        for key in keys:
            localconf[key] = self.task_config[key]
        self.get_fv3ens_dict(localconf)
```

Having them in `self.get_fv3ens_dict` is not really a good solution as it makes that method bring in the entire global environment. One should know exactly what a method requires and pass that explicitly.  One could say it is the same argument that people frown on using global variables in Fortran.

Reply:
Thanks for the guidance.  Back to refactoring.

Reply:
Create, populate, and pass `localconf` to `get_fv3ens_dict`.   Change made in both `atm_analysis.py` and `atmens_analysis.py`.  Done at [`a2ab183`](https://github.com/RussTreadon-NOAA/global-workflow/commit/a2ab18339287dc73cb36d269e86b641d54445f55)

Reply:
perfect @RussTreadon-NOAA 
This way, when the method runs, the logger will print out exactly what is going in and coming out.
Thank you~